test/utils.test.ts

```diff
@@ -78,6 +78,69 @@ describe("utils", () => {
     )
   })
 
+  it("normalize addresses to checksum, given eth_mainnet chainId", () => {
+    const chainId = 1
+    const ethMainnet = [
+      "0x88021160c5C792225E4E5452585947470010289d",
+      "0x27b1FdB04752bBc536007a920D24ACB045561c26",
+      "0x52908400098527886e0f7030069857D2e4169EE7",
+      "0x5aaeB6053f3E94C9b9A09f33669435e7Ef1bEAed",
+      "0x8617E340b3d01FA5F11F306f4090FD50E238070d",
+      "0xd1220a0CF47C7B9Be7A2E6ba89F429762E7B9Adb",
+      "0xdBf03b407c01e7cD3CBea99509d93f8dDDC8C6fB",
+      "0xDe709F2102306220921060314715629080E2fb77",
+      "0xfb6916095Ca1dF60bB79cE92ce3ea74C37c5D359",
+    ]
+    ethMainnet.forEach((testAddress) => {
+      expect(toChecksumAddress(testAddress.toLowerCase(), chainId)).toEqual(
+        testAddress
+      )
+    })
+  })
```

Comment:
Can we add one more set of test cases, without the chain id? Once we have that I think the rest of this looks 👌

Reply:
Actually just realized those are already there, I was looking at the latest diff.

On closer reading of EIP-1191 though we should be whitelisting the chain id; i.e., we should only accept chain ids that have opted in, otherwise we should skip the prefix (this fixes the fact that passing the Ethereum mainnet chain id shouldn't change the checksum used for mainnet, which would produce invalid mainnet addresses).

Reply:
Planning on fixing this up this weekend. Been crazy at work past week or so.